packages/grafana-ui/src/components/DataLinks/DataLinkInput.tsx

```diff
@@ -198,6 +202,25 @@ export const DataLinkInput = memo(
       return precedingChar;
     };
 
+    const getUrlLength = () => {
+      const input: HTMLSpanElement | null = document.getElementById('data-link-input')!;
```

Comment:
This `input` value uses a non-null assertion (!) on `document.getElementById('data-link-input')`, but then also asserts that the type of said value **_could_** include a possible null. Is that intentional?

Reply:
Looks like this might be copied from above? But I question that as well. 🤔

Reply:
@baldm0mma Thank you for pointing that out! That part of the `getUrlLength` method was indeed copied from the `getCharactersAroundCaret` method. Given your observation, it looks like the use of the non-null assertion (!) could be improved for better code safety.

Reply:
I have removed `getUrlLength` method. How about changing to something like this in `getCharactersAroundCaret`.

```
const input: HTMLSpanElement | null = document.getElementById('data-link-input');
  if (!input) {
    return '';
  }
```